app/black_spots/tasks/generate_training_input.py

```diff
@@ -0,0 +1,361 @@
+import fiona
+import tarfile
+import shutil
+from collections import namedtuple
+from django.core.files import File
+
+from celery import shared_task
+from celery.utils.log import get_task_logger
+
+from shapely.geometry import shape, Point
+from shapely.ops import transform
+
+from black_spots.models import (
+    BlackSpotRecordsFile, RoadSegmentsShapefile, BlackSpotTrainingCsv, ForecastTrainingCsv
+)
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger()
+logger = get_task_logger(__name__)
+
+INTERSECTION_BUFFER_UNITS = int(os.getenv('INTERSECTION_BUFFER_UNITS', '5'))
+MAX_LINE_UNITS = int(os.getenv('MAX_LINE_UNITS', '200'))
+RECORD_PROJECTION = os.getenv('RECORD_PROJECTION', 'epsg:4326')
+TIMEZONE = pytz.timezone(os.getenv('TIMEZONE', 'Asia/Manila'))
+MATCH_TOLERANCE = int(os.getenv('MATCH_TOLERANCE', '5'))
+RECORD_COL_ID = os.getenv('RECORD_COL_ID', 'record_id')
+RECORD_COL_X = os.getenv('RECORD_COL_X', 'lon')
+RECORD_COL_Y = os.getenv('RECORD_COL_Y', 'lat')
+RECORD_COL_OCCURRED = os.getenv('RECORD_COL_OCCURRED', 'occurred_from')
+RECORD_COL_SEVERE = os.getenv('RECORD_COL_SEVERE', 'Severity')
+RECORD_COL_SEVERE_VALS = os.getenv('RECORD_COL_SEVERE_VALS', 'Fatal,Injury')
+RECORD_COL_PRECIP = os.getenv('RECORD_COL_PRECIP', 'weather')
+RECORD_COL_PRECIP_VALS = os.getenv('RECORD_COL_PRECIP_VALS', 'rain,hail,sleet,snow,thunderstorm,tornado')
+TILE_MAX_UNITS = int(os.getenv('TILE_MAX_UNITS', '3000'))
+COMBINED_SEGMENTS_SHP_NAME = os.getenv('COMBINED_SEGMENTS_SHP_NAME', 'combined_segments.shp')
+ROAD_PROJECTION = os.getenv('ROAD_PROJECTION', 'epsg:32651')
+
+
+@shared_task
+def get_training_input(segments_shp_uuid, records_csv_uuid):
+    """Generate training input
+
+    Output to training csv to django-storages
+
+    Args:
+    :param segments_tar_uuid: (str) UUID for the RoadSegmentsShapefile to use
+    :param records_csv_uuid: (str) UUID for the BlackSpotRecordsFile
+
+    Returns:
+    TrainingCsvs: (namedtuple ['blackspots', 'forecast'])  tuple containing the
+                  BlackSpotTrainingCsv and ForecastTrainingCsv uuids
+    """
+
+    # Use passed uuid
+    with BlackSpotRecordsFile.objects.get(uuid=records_csv_uuid).csv as records_csv:
+        records, min_occurred, max_occurred = read_records(
+            records_csv, {'init': ROAD_PROJECTION}, {'init': RECORD_PROJECTION}, TIMEZONE,
+            RECORD_COL_ID, RECORD_COL_X, RECORD_COL_Y,
+            RECORD_COL_OCCURRED, RECORD_COL_SEVERE, RECORD_COL_SEVERE_VALS,
+            RECORD_COL_PRECIP, RECORD_COL_PRECIP_VALS
+        )
+    records_with_precip = [record for record in records if record['precip']]
+    records_without_precip = [record for record in records if not record['precip']]
+
+    # segments shapefile
+    tar_output_dir = tempfile.mkdtemp()
+    try:
+        shp_tar = RoadSegmentsShapefile.objects.get(uuid=segments_shp_uuid).shp_tgz.path
+        with tarfile.open(shp_tar, "r:gz") as tar:
```

Comment:
Probably switch this to single quotes, since every other string in the file uses single quotes.